docs/standard/exceptions.md

```diff
@@ -0,0 +1,528 @@
+
+C#
+```
+using System;
+using System.IO;
+
+public class ProcessFile
+{
+   public static void Main()
+      {
+      FileStream fs = null;
+      try   
+      {
+         //Opens a text tile.
+         fs = new FileStream(@"C:\temp\data.txt", FileMode.Open);
+         StreamReader sr = new StreamReader(fs);
+         string line;
+
+         //A value is read from the file and output to the console.
+         line = sr.ReadLine();
+         Console.WriteLine(line);
+      }
+      catch(FileNotFoundException e)
+      {
+         Console.WriteLine("[Data File Missing] {0}", e);
+         throw new FileNotFoundException(@"[data.txt not in c:\temp directory]",e);
+      }
+      finally
+      {
+         if (fs != null)
+            fs.Dispose();
+      }
+   }
+}
+```
+
+## How to create user-defined exceptions
+
+.NET provides a hierarchy of exception classes ultimately derived from the base class [Exception](https://docs.microsoft.com/dotnet/core/api/System.Exception). However, if none of the predefined exceptions meets your needs, you can create your own exception classes by deriving from the **Exception** class.
+
+When creating your own exceptions, end the class name of the user-defined exception with the word "Exception," and implement the three common constructors, as shown in the following example. The example defines a new exception class named `EmployeeListNotFoundException`. The class is derived from **Exception** and includes three constructors.
+
+C#
+```
+using System;
+
+public class EmployeeListNotFoundException: Exception
+{
+    public EmployeeListNotFoundException()
+    {
+    }
+
+    public EmployeeListNotFoundException(string message)
+        : base(message)
+    {
+    }
+
+    public EmployeeListNotFoundException(string message, Exception inner)
+        : base(message, inner)
+    {
+    }
+}
+```
+
+> **Note** 
+> 
+> In situations where you are using remoting, you must ensure that the metadata for any user-defined exceptions is available at the server (callee) and to the client (the proxy object or caller). For more information, see [Best practices for exceptions](#best-practices-for-exceptions).
+
+## Best practices for exceptions
+
+A well-designed app handles exceptions and errors to prevent app crashes. This section describes best practices for handling and creating exceptions.
+
+### Use try/catch/finally blocks
+
+Use **try**/**catch** blocks around code that can potentially generate an exception, and use a **finally** block to clean up resources, if necessary.
```

Comment:
I would re-work just this sentence to emphasize 'finally'. You really only want to catch an exception if you can recover. You want a finally block for any resource cleanup, whether you can recover or not.

In practice, that means you'll use finally a lot more than catch.